src/os/bluestore/BlueStore.cc

```diff
@@ -8314,8 +8314,12 @@ void BlueStore::_dump_extent_map(ExtentMap &em, int log_level)
   }
 }
 
-void BlueStore::_dump_transaction(Transaction *t, int log_level)
+void BlueStore::_dump_transaction(Transaction* const t, const int log_level)
 {
+  if (! cct->_conf->subsys.should_gather(ceph_subsys_bluestore, log_level)) {
```

Comment:
Yep, there's an implicit block from dout to dendl; no need to check should_gather explicitly